### PR TITLE
Feature add git flow instructions

### PR DESCRIPTION
**Pull request by barthr**
- **PR Description**

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Comment by barthr** [comment]
This PR resolves: https://github.com/jesseduffield/lazygit/issues/2606

File: README.md [hunk 1]
```diff
@@ -297,6 +297,12 @@ See the [docs](docs/Custom_Command_Keybindings.md)
 
 ![Interactive Rebasing](../assets/rebase.gif)
 
+### Git flow support
+
+[Git Flow](https://github.com/nvie/gitflow) is a Git branching model proposed by Vincent Driessen that streamlines and standardizes the process of development, release, and maintenance. 
+It introduces a strict branching structure with specific roles: master or main for production-ready code, develop for active development, and feature/*, release/*, hotfix/* branches for features, releases, and bug fixes, respectively. 
+To learn more, refer to Vincent Driessen's original [post](https://nvie.com/posts/a-successful-git-branching-model/).
+
```
Comment (jesseduffield) [inline README.md:305]: I'd mention that Lazygit supports git flow, but only if you already have git flow installed

Comment (barthr) [reply README.md:305]: That makes sense, I will add that

**Comment by jesseduffield** [comment]
I pushed a change which makes the section a little more concise. Thanks for the PR!

**Pull request merged**